temporalio/worker/_workflow_instance.py

```diff
@@ -1954,10 +2011,75 @@ def _enhanced_stack_trace(self) -> temporalio.api.sdk.v1.EnhancedStackTrace:
+        query_definitions = [
+            temporalio.api.sdk.v1.WorkflowInteractionDefinition(
+                name=qd.name if qd.name is not None else "",
+                description=qd.description if qd.description is not None else "",
+            )
+            for qd in self._queries.values()
+        ]
+        query_definitions.sort(key=lambda qd: qd.name)
+        signal_definitions = [
+            temporalio.api.sdk.v1.WorkflowInteractionDefinition(
+                name=sd.name if sd.name is not None else "",
+                description=sd.description if sd.description is not None else "",
+            )
+            for sd in self._signals.values()
+        ]
+        signal_definitions.sort(key=lambda sd: sd.name)
+        update_definitions = [
+            temporalio.api.sdk.v1.WorkflowInteractionDefinition(
+                name=ud.name if ud.name is not None else "",
+                description=ud.description if ud.description is not None else "",
+            )
+            for ud in self._updates.values()
+        ]
+        update_definitions.sort(key=lambda ud: ud.name)
+        wf_def = temporalio.api.sdk.v1.WorkflowDefinition(
+            type=self._info.workflow_type,
+            query_definitions=query_definitions,
+            signal_definitions=signal_definitions,
+            update_definitions=update_definitions,
+        )
+        cur_details = self.workflow_get_current_details()
+        return temporalio.api.sdk.v1.WorkflowMetadata(
+            definition=wf_def, current_details=cur_details
+        )
+
+    def _timer_impl(
+        self,
+        delay: float,
+        callback: Callable[..., Any],
+        *args: Any,
+        context: Optional[contextvars.Context] = None,
+        options: Optional[_TimerOptions] = None,
+    ):
+        self._assert_not_read_only("schedule timer")
+        # Delay must be positive
+        if delay < 0:
+            raise RuntimeError("Attempting to schedule timer with negative delay")
+
+        # Create, schedule, and return
+        seq = self._next_seq("timer")
+        # If options aren't explicitly passed, attempt to fetch them from the class field,
+        # erasing them afterward. Support callers who cannot call this directly because they
+        # rely on asyncio functions.
+        if options is None:
+            options = self._next_timer_options
+            self._next_timer_options = None
+        handle = _TimerHandle(
+            seq, self.time() + delay, options, callback, args, self, context
+        )
+        handle._apply_start_command(self._add_command(), delay)
+        self._pending_timers[seq] = handle
+        return handle
+
     #### asyncio.AbstractEventLoop function impls ####
     # These are in the order defined in CPython's impl of the base class. Many
     # functions are intentionally not implemented/supported.
 
+    # TODO/Review: This doesn't appear to implement any base class fn and isn't called anywhere?
```

Comment:
This is defined in the abstract impl at https://github.com/python/cpython/blob/v3.13.1/Lib/asyncio/events.py#L274, invoked on `TimerHandle.cancel()` which means I think this is invoked when timer is no longer needed, such as in a timeout situation where it completes before timeout. I'd have to test to be sure.